File: Finales/final_2dafecha.py
def nuevaSecuencia (s: list[int], t: list[int]) -> list[int]:
    res: list[int] = []
    maximo: int = 0
    subsecuencia: list[int] = []
    e: int = 0
    for i in range (0,len(s),1):
        if s[i] <= len(t):
            for n in range (0,s[i]+1,1):
                subsecuencia.append(t[n])
            print(subsecuencia)    
            while e < len(subsecuencia):
                 if subsecuencia[e] > maximo:
                     maximo = subsecuencia[e]
                 e+=1
            res.append(maximo)
            maximo = 0
            e=0
            subsecuencia = []
        else:
            res.append(t[len(t)-1])    
    return res                     
                         
s=[4,3,2,1,7]
t=[1,2,3,4,5]
#resultado esperado = [5,4,3,2,5]
print(nuevaSecuencia(s,t))

File: Finales/final_2dafecha.hs
nuevaSecuencia :: [Int] -> [Int] -> [Int]
nuevaSecuencia (x:[]) (y:ys) | x <= longitud (y:ys) =[maximo (subsecuencia x (y:ys))]
                             | otherwise = [ultimo (y:ys)]
nuevaSecuencia (x:xs) (y:ys) | x <= longitud (y:ys) =[maximo (subsecuencia x (y:ys))] ++ nuevaSecuencia (xs) (y:ys)
                             | otherwise = [ultimo (y:ys)] ++ nuevaSecuencia (xs) (y:ys)


ultimo :: [Int] -> Int
ultimo [x] = x
ultimo (x:xs) = ultimo (xs)

longitud :: [Int] -> Int
longitud [x] = 1
longitud (x:xs) = 1 + longitud (xs)

maximo :: [Int] -> Int
maximo [x] = x
maximo (x:y:[]) | x>=y =x
                | x<y =y
maximo (x:y:xs) | x>=y =maximo (x:xs)
                | x<y =maximo (y:xs)

subsecuencia :: Int -> [Int] -> [Int]
subsecuencia n (y:ys) | n==0 =[y]
                      | n>0 = [y] ++ subsecuencia (n-1) (ys)                 